### PR TITLE
Download frames in order they appear

### PR DESCRIPTION
**Pull request by n-wach**
Frames were being downloaded out of order because we used a `set` of hashes (unordered). Now we use a `OrderedDict`

Old CVC would link as files were downloaded. Now we do the same. This involved quite a bit of refactoring because of our reduction to unique hashes.

### CODE REVIEW

File: FLIR/conservator/local_dataset.py [hunk 1]
```diff
@@ -355,69 +382,61 @@ def download(
...
-
-                if include_analytics and ("analyticsMd5" in frame):
-                    md5 = frame["analyticsMd5"]
-                    hashes_required.add(md5)
-
-                    name = f"video-{video_id}-frame-{frame_index:06d}-{dataset_frame_id}.tiff"
-                    path = os.path.join(self.analytics_path, name)
-                    cache_path = self.get_cache_path(md5)
-                    links.append((cache_path, path))
+        # dict stores unique keys in order of insertion. this maps hash -> [links]
```
Comment (rlc2) [inline FLIR/conservator/local_dataset.py:385]: This is true in Python 3.7+, is that the minimum required version in setup.py?

Comment (rlc2) [reply FLIR/conservator/local_dataset.py:385]: In main it has `python_requires=">=3.6"`

Comment (n-wach) [reply FLIR/conservator/local_dataset.py:385]: Should we bump it?

Or replace with `list` of `(hash, links)` and `set` of `hash`.

Comment (rlc2) [reply FLIR/conservator/local_dataset.py:385]: There is also collections.OrderedDict

**Review by rlc2 — APPROVED** [review]

**Pull request merged**